src/plugins/auto/bind_multi_schedule.cpp

```diff
@@ -0,0 +1,186 @@
+    bool flag = false;
+    while (idleWorkerRequests.try_pop(workerRequestPtr)) {
+        if (flag && workerRequestPtr == headWorker)
+            break;
+        if (!flag) {
+            headWorker = workerRequestPtr;
+            flag = true;
+        }
+        IdleGuard<NotBusyWorkerRequests> idleGuard{workerRequestPtr, idleWorkerRequests};
+        if (_sharedRequest._ptr.get() == workerRequestPtr->_inferRequest._ptr.get()) {
+            _thisWorkerInferRequest = workerRequestPtr;
+            {
+                auto capturedTask = std::move(inferPipelineTask);
+                capturedTask();
+            }
+            idleGuard.Release();
+            return true;
+        }
+    }
+    return false;
+}
+
+void BinderMultiSchedule::run(IE::Task inferPipelineTask) {
+    if (_thisWorkerInferRequest) {
+        auto capturedTask = std::move(inferPipelineTask);
+        capturedTask();
+    } else {
+        ScheduleToWorkerInferRequest(std::move(inferPipelineTask), _thisPreferredDeviceName);
+    }
+}
+
+BinderMultiSchedule::~BinderMultiSchedule() {
+}
+
+IInferPtr BinderMultiSchedule::CreateInferRequestImpl(
+    const std::vector<std::shared_ptr<const ov::Node>>& inputs,
+    const std::vector<std::shared_ptr<const ov::Node>>& outputs) {
+    auto num = _numRequestsCreated++;
+    size_t sum = 0;
+    SoInfer request_to_share_blobs_with;
+    IE::RemoteContext::Ptr ctx = nullptr;
+    // borrowing device-specific blobs from the underlying requests for the device-agnostic, user-facing requests
+    // this allows to potentially save on the data-copy later (if the requests are scheduled in the same order)
+    for (const auto& device : _multiSContext->_devicePrioritiesInitial) {
+        auto& dev_requests = _workerRequests[device.deviceName];
+        if ((num - sum) < dev_requests.size()) {
+            request_to_share_blobs_with = dev_requests.at(num - sum)._inferRequest;
+            break;
+        }
+        sum += dev_requests.size();
+    }
+    auto syncImpl = std::make_shared<MultiDeviceInferRequest>(inputs, outputs, request_to_share_blobs_with);
+    return syncImpl;
+}
+
+IInferPtr BinderMultiSchedule::CreateInferRequestImpl(IE::InputsDataMap networkInputs,
+    IE::OutputsDataMap networkOutputs) {
+    auto num = _numRequestsCreated++;
+    SoInfer request_to_share_blobs_with;
+    size_t sum = 0;
+    IE::RemoteContext::Ptr ctx = nullptr;
+    // borrowing device-specific blobs from the underlying requests for the device-agnostic, user-facing requests
+    // this allows to potentially save on the data-copy later (if the requests are scheduled in the same order)
+    for (const auto& device : _multiSContext->_devicePrioritiesInitial) {
+        auto& dev_requests = _workerRequests[device.deviceName];
+        if ((num - sum) < dev_requests.size()) {
+            request_to_share_blobs_with = dev_requests.at(num - sum)._inferRequest;
+            break;
+        }
+        sum += dev_requests.size();
```

Comment:
is bind schedule coverd the case when customer create more request than expect

Reply:
this binder schedule, is used for AXG/DPG in advanced mode, assuming user will follow the optimal infer request. for future improvement, we can continue to improve this schedule to cover more complicate cases